internal/app/maintenance/server/server.go

```diff
@@ -83,6 +83,14 @@ func (s *Server) ApplyConfiguration(ctx context.Context, in *machine.ApplyConfig
 		},
 	}
 
+	if in.DryRun {
+		reply.Messages[0].ModeDetails = `dry run summary:
+node is running in mainenance mode
+applied config will be used after the reboot (skipped in dry run)`
```

Comment:
This is not quite true (`after a reboot`), as for disk image based setups there will be no reboot.

I would rather say that there's no existing machine configuration to compare with.